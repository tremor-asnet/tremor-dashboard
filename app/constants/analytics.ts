/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
