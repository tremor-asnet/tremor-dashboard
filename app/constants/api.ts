import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
