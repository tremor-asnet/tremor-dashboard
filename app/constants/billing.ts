export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
