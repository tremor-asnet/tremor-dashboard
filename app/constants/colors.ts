type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
