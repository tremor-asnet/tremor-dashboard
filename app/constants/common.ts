export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
