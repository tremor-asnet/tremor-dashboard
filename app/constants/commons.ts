export const ITEM_ACTION_PROJECT = [
  {
    key: "act",
    label: "Action",
  },
  {
    key: "anotherAct",
    label: "Another action",
  },
  {
    key: "somethingElse",
    label: "Something else here",
  },
];

export const ITEM_ACTION_SALES_DATE = [
  {
    key: "last7Days",
    label: "Last 7 days",
  },
  {
    key: "lastWeek",
    label: "Last week",
  },
  {
    key: "last30Days",
    label: "Last 30 days",
  },
];

export const URL_AVATAR_PROJECT = [
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
];

// Social link
export const SOCIAL_LINK = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};

export const REMEMBER_ME = "remember-me";

export const listOption: ListOptionProps[] = [
  { option: "Paid" },
  { option: "Refunded" },
  { option: "Canceled" },
];
