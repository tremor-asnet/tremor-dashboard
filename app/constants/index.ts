export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
