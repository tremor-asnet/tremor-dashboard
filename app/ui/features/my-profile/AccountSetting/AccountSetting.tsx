"use client";
import { Switch, Text } from "@tremor/react";

// Types
import { PlatformSetting, AccountSettingData } from "@/types";

// Contexts
import { useContext } from "react";

// Themes
import { ThemeContext } from "@/context/theme";

interface AccountSettingProps {
  accountSettingFields: PlatformSetting[];
  accountSettingData: AccountSettingData;
}

const AccountSetting = ({
  accountSettingFields,
  accountSettingData,
}: AccountSettingProps) => {
  const { isDarkTheme } = useContext(ThemeContext);
  const color = isDarkTheme ? "white" : "zinc";

  return (
    <>
      <Text className="text-xs leading-tight dark:text-dark-romance opacity-100 uppercase no-underline text-[#7b809a] font-bold m-0 pt-4">
        Account
      </Text>
      {accountSettingFields.map(({ label, field }: PlatformSetting) => (
        <div
          className="flex items-center space-x-3 mt-1 py-3"
          key={label}
          data-testid="account-setting">
          <Switch
            tabIndex={2}
            id="switch"
            name="switch"
            color={color}
            className="flex justify-center items-center dark:text-dark-romance"
            defaultChecked={accountSettingData[field]}
          />
          <Text className="text-secondary font-normal dark:text-dark-romance">
            {label}
          </Text>
        </div>
      ))}
    </>
  );
};

export default AccountSetting;
