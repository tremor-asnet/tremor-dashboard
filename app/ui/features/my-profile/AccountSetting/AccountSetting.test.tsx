import { fireEvent, render, screen } from "@testing-library/react";

// Contexts
import { ThemeContext, ThemeProvider } from "@/context/theme";

// Components
import AccountSetting from "./AccountSetting";

// Constants
import { ACCOUNT_SETTING_FIELDS, ACCOUNT_SETTING_DATA } from "@/constants";

describe("AccountSetting component", () => {
  const props = {
    accountSettingFields: ACCOUNT_SETTING_FIELDS,
    accountSettingData: ACCOUNT_SETTING_DATA,
  };

  const propsEmptyData = {
    accountSettingFields: [],
    accountSettingData: {},
  };

  it("Should render AccountSetting snapshot correctly", () => {
    const { getByTestId, container } = render(
      <ThemeProvider>
        <ThemeContext.Consumer>
          {({ isDarkTheme, toggleTheme }) => (
            <>
              <AccountSetting {...props} />
              <button
                data-testid="toggle-theme"
                onClick={() =>
                  toggleTheme()
                }>{`isDarkTheme to ${isDarkTheme}`}</button>
            </>
          )}
        </ThemeContext.Consumer>
      </ThemeProvider>,
    );
    const themeButton = getByTestId("toggle-theme");

    fireEvent.click(themeButton);
    expect(themeButton.textContent).toBe("isDarkTheme to true");

    fireEvent.click(themeButton);
    expect(themeButton.textContent).toBe("isDarkTheme to false");

    expect(container).toMatchSnapshot();
  });

  it("Should render AccountSetting in case empty data", () => {
    render(<AccountSetting {...propsEmptyData} />);

    expect(screen.queryByTestId("account-setting")).toBeNull();
  });
});
