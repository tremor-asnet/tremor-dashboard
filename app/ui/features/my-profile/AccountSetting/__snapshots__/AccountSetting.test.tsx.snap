// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSetting component Should render AccountSetting snapshot correctly 1`] = `
<div>
  <p
    class="text-tremor-default text-xs leading-tight dark:text-dark-romance opacity-100 uppercase no-underline text-[#7b809a] font-bold m-0 pt-4"
  >
    Account
  </p>
  <div
    class="flex items-center space-x-3 mt-1 py-3"
    data-testid="account-setting"
  >
    <div
      class="flex flex-row items-center justify-start"
    >
      <div
        class="flex justify-center items-center dark:text-dark-romance"
        tabindex="2"
      >
        <input
          class="tremor-Switch-input absolute w-5 h-5 cursor-pointer left-0 top-0 opacity-0"
          name="switch"
          type="checkbox"
        />
        <button
          aria-checked="false"
          class="tremor-Switch-switch w-10 h-5 group relative inline-flex shrink-0 cursor-pointer items-center justify-center rounded-tremor-full focus:outline-none"
          data-headlessui-state=""
          id="switch"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="tremor-Switch-sr-only sr-only"
          >
            Switch 
            off
          </span>
          <span
            aria-hidden="true"
            class="tremor-Switch-background bg-tremor-border dark:bg-dark-tremor-border pointer-events-none absolute mx-auto h-3 w-9 rounded-tremor-full transition-colors duration-100 ease-in-out"
          />
          <span
            aria-hidden="true"
            class="tremor-Switch-round translate-x-0 bg-tremor-border dark:bg-dark-tremor-border border-tremor-background dark:border-dark-tremor-background pointer-events-none absolute left-0 inline-block h-5 w-5 transform rounded-tremor-full border-2 shadow-tremor-input duration-100 ease-in-out transition"
          />
        </button>
      </div>
    </div>
    <p
      class="text-tremor-default text-secondary font-normal dark:text-dark-romance"
    >
      Email me when someone follows me
    </p>
  </div>
  <div
    class="flex items-center space-x-3 mt-1 py-3"
    data-testid="account-setting"
  >
    <div
      class="flex flex-row items-center justify-start"
    >
      <div
        class="flex justify-center items-center dark:text-dark-romance"
        tabindex="2"
      >
        <input
          class="tremor-Switch-input absolute w-5 h-5 cursor-pointer left-0 top-0 opacity-0"
          name="switch"
          type="checkbox"
        />
        <button
          aria-checked="false"
          class="tremor-Switch-switch w-10 h-5 group relative inline-flex shrink-0 cursor-pointer items-center justify-center rounded-tremor-full focus:outline-none"
          data-headlessui-state=""
          id="switch"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="tremor-Switch-sr-only sr-only"
          >
            Switch 
            off
          </span>
          <span
            aria-hidden="true"
            class="tremor-Switch-background bg-tremor-border dark:bg-dark-tremor-border pointer-events-none absolute mx-auto h-3 w-9 rounded-tremor-full transition-colors duration-100 ease-in-out"
          />
          <span
            aria-hidden="true"
            class="tremor-Switch-round translate-x-0 bg-tremor-border dark:bg-dark-tremor-border border-tremor-background dark:border-dark-tremor-background pointer-events-none absolute left-0 inline-block h-5 w-5 transform rounded-tremor-full border-2 shadow-tremor-input duration-100 ease-in-out transition"
          />
        </button>
      </div>
    </div>
    <p
      class="text-tremor-default text-secondary font-normal dark:text-dark-romance"
    >
      Email me when someone answers on my post
    </p>
  </div>
  <div
    class="flex items-center space-x-3 mt-1 py-3"
    data-testid="account-setting"
  >
    <div
      class="flex flex-row items-center justify-start"
    >
      <div
        class="flex justify-center items-center dark:text-dark-romance"
        tabindex="2"
      >
        <input
          class="tremor-Switch-input absolute w-5 h-5 cursor-pointer left-0 top-0 opacity-0"
          name="switch"
          type="checkbox"
        />
        <button
          aria-checked="false"
          class="tremor-Switch-switch w-10 h-5 group relative inline-flex shrink-0 cursor-pointer items-center justify-center rounded-tremor-full focus:outline-none"
          data-headlessui-state=""
          id="switch"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="tremor-Switch-sr-only sr-only"
          >
            Switch 
            off
          </span>
          <span
            aria-hidden="true"
            class="tremor-Switch-background bg-tremor-border dark:bg-dark-tremor-border pointer-events-none absolute mx-auto h-3 w-9 rounded-tremor-full transition-colors duration-100 ease-in-out"
          />
          <span
            aria-hidden="true"
            class="tremor-Switch-round translate-x-0 bg-tremor-border dark:bg-dark-tremor-border border-tremor-background dark:border-dark-tremor-background pointer-events-none absolute left-0 inline-block h-5 w-5 transform rounded-tremor-full border-2 shadow-tremor-input duration-100 ease-in-out transition"
          />
        </button>
      </div>
    </div>
    <p
      class="text-tremor-default text-secondary font-normal dark:text-dark-romance"
    >
      Email me when someone mentions me
    </p>
  </div>
  <button
    data-testid="toggle-theme"
  >
    isDarkTheme to false
  </button>
</div>
`;
